lib/modules/manager/mix/extract.ts

```diff
@@ -18,7 +20,7 @@ export async function extractPackageFile(
   const deps: PackageDependency[] = [];
   const contentArr = content
     .split(newlineRegex)
-    .filter((line) => !commentMatchRegExp.test(line));
+    .map((line) => line.replace(commentMatchRegExp, ''));
```

Comment:
this seems to be a fix for a different issue. can you open a separate PR for it?

Reply:
It would seem so, but in the fixture there's this line:

```elixir
{:postgrex, "~> 0.8.1"}, # {:broadway_dashboard, "~> 0.2.2"},
```

The previous `depMatchRegExp` had a `.*}` at the end which basically made it consider the line a single dependency definition matched as `{:postgrex, "~> 0.8.1"}, # {:broadway_dashboard, "~> 0.2.2"}`.

The new `depMatchRegExp` is more strict and should match exactly one dependency definition at a time (unless it uses the `}` character in one of the options).

So to prevent it from picking up `{:broadway_dashboard, "~> 0.2.2"}` I decided to improve how comments are cut out of the input. It's not a great solution - a parser should be used instead - but it's better than the previous one.

I could keep the `.*}` but since I re-wrote the regex I don't think I should make it buggy again.

Reply:
Opened as #19142.